src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide(hidden) {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(herbivore) {
+    if (herbivore.hidden === false) {
+      herbivore.health -= 50;
+    }
+
+    if (herbivore.health <= 0) {
+      const index = Animal.alive.indexOf(herbivore);
+
+      if (index !== -1) {
+        Animal.alive.splice(index, 1);
+      }
```

Comment:
[CODE PERFOMANCE] - Instead of using `indexOf` and `splice` to remove the dead animal from the `Animal.alive` array, consider using the `filter` method to avoid nesting loops and improve performance. For example, you can filter out animals whose health is less than or equal to 0 after decreasing the health of the herbivore.